diff --git a/Makefile b/Makefile
old mode 100644
new mode 100755
index a8514826..8d370166
--- a/Makefile
+++ b/Makefile
@@ -13,11 +13,11 @@ DAEMON_SRC_DIR := daemon
 
 include filelist.mk
 
-USE_AESNI	:= yes
-USE_AVX		:= yes
+USE_AESNI	:= no
+USE_AVX		:= no
 USE_STATIC	:= no
 USE_MESHNET	:= no
-USE_UPNP	:= no
+USE_UPNP	:= yes
 
 ifeq ($(WEBSOCKETS),1)
 	NEEDED_CXXFLAGS += -DWITH_EVENTS
diff --git a/contrib/i2pd.conf b/contrib/i2pd.conf
old mode 100644
new mode 100755
index d9fe951b..84b1826b
--- a/contrib/i2pd.conf
+++ b/contrib/i2pd.conf
@@ -8,10 +8,10 @@
 
 ## Tunnels config file
 ## Default: ~/.i2pd/tunnels.conf or /var/lib/i2pd/tunnels.conf
-# tunconf = /var/lib/i2pd/tunnels.conf
+tunconf = /etc/i2pd_tunnels.conf
 
 ## Where to write pidfile (don't write by default)
-# pidfile = /var/run/i2pd.pid
+pidfile = /var/run/i2pd.pid
 
 ## Logging configuration section
 ## By default logs go to stdout with level 'info' and higher
@@ -20,21 +20,21 @@
 ##  * stdout - print log entries to stdout
 ##  * file - log entries to a file
 ##  * syslog - use syslog, see man 3 syslog
-# log = file
+log = file
 ## Path to logfile (default - autodetect)
-# logfile = /var/log/i2pd.log
+logfile = /var/log/i2pd.log
 ## Log messages above this level (debug, *info, warn, error, none)
 ## If you set it to none, logging will be disabled
-# loglevel = info
+loglevel = warn
 
 ## Path to storage of i2pd data (RI, keys, peer profiles, ...)
 ## Default: ~/.i2pd or /var/lib/i2pd
 # datadir = /var/lib/i2pd
 
 ## Daemon mode. Router will go to background after start
-# daemon = true
+daemon = true
 ## Run as a service. Router will use system folders like ‘/var/lib/i2pd’
-# service = true
+service = true
 
 ## Specify a family, router belongs to (default - none)
 # family =
