diff -Naur p7zip-16.02-orig/makefile.hpux-gcc p7zip-16.02/makefile.hpux-gcc
--- p7zip-16.02-orig/makefile.hpux-gcc	2017-12-31 08:52:29.900879699 +0300
+++ p7zip-16.02/makefile.hpux-gcc	2017-12-31 09:03:16.191478384 +0300
@@ -2,7 +2,7 @@
 # makefile for HP-UX 11
 #
 
-OPTFLAGS=-O -s
+OPTFLAGS=-O2 -msse2 -s -Wcheck -fstack-security-check
 
 ALLFLAGS=${OPTFLAGS} \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
diff -Naur p7zip-16.02-orig/makefile.linux_amd64 p7zip-16.02/makefile.linux_amd64
--- p7zip-16.02-orig/makefile.linux_amd64	2017-12-31 08:52:29.900879699 +0300
+++ p7zip-16.02/makefile.linux_amd64	2017-12-31 09:03:10.227587858 +0300
@@ -1,5 +1,5 @@
 
-OPTFLAGS=-O -s
+OPTFLAGS=-O2 -msse2 -s -Wcheck -fstack-security-check
 
 # use "-m32" to have a 32bits executable
 ALLFLAGS=-m64 ${OPTFLAGS} -pipe \
diff -Naur p7zip-16.02-orig/makefile.linux_any_cpu p7zip-16.02/makefile.linux_any_cpu
--- p7zip-16.02-orig/makefile.linux_any_cpu	2017-12-31 08:52:29.900879699 +0300
+++ p7zip-16.02/makefile.linux_any_cpu	2017-12-31 09:00:38.009074525 +0300
@@ -2,7 +2,7 @@
 # makefile for Linux (x86, PPC, alpha ...)
 #
 
-OPTFLAGS=-O -s
+OPTFLAGS=-O2 -msse2 -s -Wcheck -fstack-security-check
 
 ALLFLAGS=${OPTFLAGS} -pipe \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
diff -Naur p7zip-16.02-orig/makefile.linux_any_cpu_gcc_4.X p7zip-16.02/makefile.linux_any_cpu_gcc_4.X
--- p7zip-16.02-orig/makefile.linux_any_cpu_gcc_4.X	2017-12-31 08:52:29.900879699 +0300
+++ p7zip-16.02/makefile.linux_any_cpu_gcc_4.X	2017-12-31 08:59:20.066302543 +0300
@@ -2,7 +2,7 @@
 # makefile for Linux (x86, PPC, alpha ...)
 #
 
-OPTFLAGS=-O -s
+OPTFLAGS=-O2 -msse2 -s -Wcheck -fstack-security-check
 
 ALLFLAGS=${OPTFLAGS} -pipe \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
diff -Naur p7zip-16.02-orig/makefile.linux_other p7zip-16.02/makefile.linux_other
--- p7zip-16.02-orig/makefile.linux_other	2017-12-31 08:52:29.900879699 +0300
+++ p7zip-16.02/makefile.linux_other	2017-12-31 08:56:55.388032169 +0300
@@ -2,7 +2,7 @@
 # makefile for Linux (other)
 #
 
-OPTFLAGS=-O -s
+OPTFLAGS=-O2 -msse2 -s -Wcheck -fstack-security-check
 
 ALLFLAGS=${OPTFLAGS} -pipe \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
diff -Naur p7zip-16.02-orig/makefile.linux_x32 p7zip-16.02/makefile.linux_x32
--- p7zip-16.02-orig/makefile.linux_x32	2017-12-31 08:52:29.932879699 +0300
+++ p7zip-16.02/makefile.linux_x32	2017-12-31 08:56:37.796383591 +0300
@@ -1,5 +1,5 @@
 
-OPTFLAGS=-O -s
+OPTFLAGS=-O2 -msse2 -s -Wcheck -fstack-security-check
 
 # use "-m32" to have a 32bits executable
 ALLFLAGS=-mx32 ${OPTFLAGS} -pipe \
diff -Naur p7zip-16.02-orig/makefile.linux_x86_asm_gcc_4.X p7zip-16.02/makefile.linux_x86_asm_gcc_4.X
--- p7zip-16.02-orig/makefile.linux_x86_asm_gcc_4.X	2017-12-31 08:52:29.908879699 +0300
+++ p7zip-16.02/makefile.linux_x86_asm_gcc_4.X	2017-12-31 08:59:28.489152151 +0300
@@ -3,7 +3,7 @@
 #
 
 # OPTFLAGS=-mfpmath=sse -Ofast -flto -march=native -funroll-loops -s
-OPTFLAGS=-O -s
+OPTFLAGS=-O2 -msse2 -s -Wcheck -fstack-security-check
 
 ALLFLAGS=${OPTFLAGS} -pipe -m32 \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
diff -Naur p7zip-16.02-orig/makefile.linux_x86_asm_gcc_4.X_fltk p7zip-16.02/makefile.linux_x86_asm_gcc_4.X_fltk
--- p7zip-16.02-orig/makefile.linux_x86_asm_gcc_4.X_fltk	2017-12-31 08:52:29.908879699 +0300
+++ p7zip-16.02/makefile.linux_x86_asm_gcc_4.X_fltk	2017-12-31 08:59:39.777698653 +0300
@@ -2,7 +2,7 @@
 # makefile for Linux (x86, PPC, alpha ...)
 #
 
-OPTFLAGS=-O -s
+OPTFLAGS=-O2 -msse2 -s -Wcheck -fstack-security-check
 
 ALLFLAGS=${OPTFLAGS} -m32 -pipe \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
diff -Naur p7zip-16.02-orig/makefile.machine p7zip-16.02/makefile.machine
--- p7zip-16.02-orig/makefile.machine	2017-12-31 08:52:29.932879699 +0300
+++ p7zip-16.02/makefile.machine	2017-12-31 08:59:06.010297492 +0300
@@ -1,7 +1,7 @@
 
 
 # OPTFLAGS=-Ofast -flto -march=native -funroll-loops -s
-OPTFLAGS=-O -s
+OPTFLAGS=-O2 -msse2 -s -Wcheck -fstack-security-check
 
 # use "-m32" to have a 32bits executable
 ALLFLAGS=-m64 ${OPTFLAGS} -pipe \
diff -Naur p7zip-16.02-orig/makefile.macosx_gcc_32bits_asm p7zip-16.02/makefile.macosx_gcc_32bits_asm
--- p7zip-16.02-orig/makefile.macosx_gcc_32bits_asm	2017-12-31 08:52:29.908879699 +0300
+++ p7zip-16.02/makefile.macosx_gcc_32bits_asm	2017-12-31 08:56:26.076029745 +0300
@@ -1,5 +1,5 @@
 
-OPTFLAGS=-O
+OPTFLAGS=-O2 -msse2 -s -Wcheck -fstack-security-check
 
 # don't use -DNDEBUG in order to use MacOSX wxWidgets
 
diff -Naur p7zip-16.02-orig/makefile.netware_asm_gcc_3.X p7zip-16.02/makefile.netware_asm_gcc_3.X
--- p7zip-16.02-orig/makefile.netware_asm_gcc_3.X	2017-12-31 08:52:29.900879699 +0300
+++ p7zip-16.02/makefile.netware_asm_gcc_3.X	2017-12-31 08:59:34.308769292 +0300
@@ -5,7 +5,7 @@
 CROSSPREFIX=i586-netware-
 BINSUFFIX=.nlm
 
-OPTFLAGS=-O -s
+OPTFLAGS=-O2 -msse2 -s -Wcheck -fstack-security-check
 
 ALLFLAGS=${OPTFLAGS} \
 	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
diff -Naur p7zip-16.02-orig/makefile.openbsd_no_port p7zip-16.02/makefile.openbsd_no_port
--- p7zip-16.02-orig/makefile.openbsd_no_port	2017-12-31 08:52:29.900879699 +0300
+++ p7zip-16.02/makefile.openbsd_no_port	2017-12-31 09:03:59.591281486 +0300
@@ -2,7 +2,7 @@
 # makefile for OpenBSD (x86, PPC, alpha ...)
 #
 
-OPTFLAGS=-O -s
+OPTFLAGS=-O2 -msse2 -s -Wcheck -fstack-security-check
 
 ALLFLAGS=${OPTFLAGS} \
         -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
